### ISSUE TITLE
Fix axe issues for Select

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly#3455]
- Select favorites stars need labels (axe issue is "Buttons must have discernible text"):
<img width="1312" alt="Screen Shot 2020-09-04 at 11 14 15 AM" src="https://user-images.githubusercontent.com/28281340/92255173-ebb5ea80-ee9f-11ea-9127-a44c2d48205e.png">

- Select roles/aria should be updated (axe issue is "Elements must only use allowed ARIA attributes"):
<img width="1312" alt="Screen Shot 2020-09-04 at 11 19 13 AM" src="https://user-images.githubusercontent.com/28281340/92255877-ed33e280-eea0-11ea-97ae-5f6ce206334e.png">

A good pattern to follow here would likely be similar to what we have in the react component and this [deque example](https://pattern-library.dequelabs.com/components/selects). Call out differences are that the `<ul>` should have role listbox, `<li>`s are role presentation, buttons are role option and `aria-selected` should be true when the item has been selected. The react markup:
<img width="759" alt="Screen Shot 2020-09-04 at 10 16 35 AM" src="https://user-images.githubusercontent.com/28281340/92255905-f4f38700-eea0-11ea-9048-56917ddff97c.png">
